src/compile/compile.ts

```diff
@@ -71,6 +71,16 @@ function assembleTopLevelModel(model: Model, topLevelProperties: TopLevelPropert
   // Config with Vega-Lite only config removed.
   const vgConfig = model.config ? stripAndRedirectConfig(model.config) : undefined;
 
+  // Assemble data before projection
+  const data = [].concat(
```

Comment:
I don't see that on master... this needs to happen before projection, unless you mean thats no longer needed... I still prefer it up here...